service/controller/resource/dnsrecord/resource.go

```diff
@@ -17,27 +16,24 @@ const (
 )
 
 type Config struct {
-	HostAzureClientSetConfig client.AzureClientSetConfig
-	Logger                   micrologger.Logger
+	CPRecordSetsClient dns.RecordSetsClient
+	Logger             micrologger.Logger
 }
 
 // Resource manages Azure resource groups.
 type Resource struct {
-	hostAzureClientSetConfig client.AzureClientSetConfig
-	logger                   micrologger.Logger
+	cpRecordSetsClient dns.RecordSetsClient
```

Comment:
The control plane azure clients can be created once, instead of once for every reconciliation loop. In those cases we inject the client when creating the resource.